examples/material_gallery/lib/demo/weathers_demo.dart

```diff
@@ -0,0 +1,495 @@
+      'packages/flutter_gallery_assets/clouds-1.png',
+      'packages/flutter_gallery_assets/ray.png',
+      'packages/flutter_gallery_assets/sun.png',
+      'packages/flutter_gallery_assets/weathersprites.json',
+      'packages/flutter_gallery_assets/weathersprites.png',
+      'packages/flutter_gallery_assets/icon-sun.png',
+      'packages/flutter_gallery_assets/icon-rain.png',
+      'packages/flutter_gallery_assets/icon-snow.png'
+    ]);
+
+    String json = await DefaultAssetBundle.of(context).loadString('packages/flutter_gallery_assets/weathersprites.json');
+    _sprites = new SpriteSheet(_images['packages/flutter_gallery_assets/weathersprites.png'], json);
+  }
+
+  void initState() {
+    super.initState();
+
+    AssetBundle bundle = DefaultAssetBundle.of(context);
+    _loadAssets(bundle).then((_) {
+      setState(() {
+        assetsLoaded = true;
+        weatherWorld = new WeatherWorld();
+      });
+    });
+  }
+
+  bool assetsLoaded = false;
+
+  WeatherWorld weatherWorld;
+
+  Widget build(BuildContext context) {
+    if (!assetsLoaded)
+      return new Text("Loading...");
+
+    return new Scaffold(
+      toolBar: new ToolBar(
+        center: new Text("Weather")
+      ),
+      body: new Material(
+        child: new Stack(
+          children: [
+            new SpriteWidget(weatherWorld),
+            new Align(
+              alignment: new FractionalOffset(0.5, 0.8),
+              child: new Row(
+                justifyContent: FlexJustifyContent.center,
+                children: [
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.sun;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.sun,
+                    icon: "packages/flutter_gallery_assets/icon-sun.png"
+                  ),
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.rain;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.rain,
+                    icon: "packages/flutter_gallery_assets/icon-rain.png"
+                  ),
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.snow;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.snow,
+                    icon: "packages/flutter_gallery_assets/icon-snow.png"
+                  )
+                ]
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+}
+
+const double _kWeatherButtonSize = 56.0;
+const double _kWeatherIconSize = 36.0;
+
+class WeatherButton extends StatelessComponent {
```

Comment:
should probably make the button a material button if we're trying to demo material+sprites. raised buttons, probably, since they're over the sprite widget.